### PR TITLE
Logging2

### PR DESCRIPTION
**Pull request by bbuzzanga**

### CODE REVIEW

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6018915&bid=19729642)

File: tools/ARIAtools/productPlot.py [hunk 1]
```diff
@@ -11,15 +11,18 @@
 import numpy as np
 from osgeo import gdal
 from mpl_toolkits.axes_grid1 import make_axes_locatable
+import logging
+from ARIAtools.logger import logger
```
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914&bid=19729641)

File: tools/ARIAtools/productPlot.py (hunk 1 above)
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/productPlot.py [hunk 2]
```diff
@@ -569,9 +568,9 @@ def main(inps=None):
     # pass number of threads for gdal multiprocessing computation
     if inps.num_threads.lower()=='all':
         import multiprocessing
-        print('User specified use of all %s threads for gdal multiprocessing'%(str(multiprocessing.cpu_count())))
+        log.info('User specified use of all %s threads for gdal multiprocessing %s', multiprocessing.cpu_count())
```
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:571]: Codacy found an issue: [Not enough arguments for logging format string](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914&bid=19729641)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6018915&bid=19729642)

File: tools/ARIAtools/productPlot.py (hunk 1 above)
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/kml2box.py [hunk 3]
```diff
@@ -9,8 +9,13 @@
 
 import os
 from osgeo import gdal
+import logging
+from ARIAtools.logger import logger
```
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/mask_util.py [hunk 4]
```diff
@@ -11,9 +11,14 @@
 import os.path as op
 import glob
 import numpy as np
+import logging
 from osgeo import gdal, ogr, gdalconst
+
 from ARIAtools.shapefile_util import open_shapefile
 from ARIAtools.vrtmanager import rasterAverage
+from ARIAtools.logger import logger
```
Comment (dbekaert) [inline tools/ARIAtools/mask_util.py:19]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/phaseMinimization.py [hunk 5]
```diff
@@ -17,6 +17,10 @@
 import pulp
 import timeit as T
 
+import logging
+from ARIAtools.logger import logger
```
Comment (dbekaert) [inline tools/ARIAtools/phaseMinimization.py:21]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

Comment (dbekaert) [inline tools/ARIAtools/phaseMinimization.py:21]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/kml2box.py (hunk 3 above)
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/productPlot.py (hunk 1 above)
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/mask_util.py (hunk 4 above)
Comment (dbekaert) [inline tools/ARIAtools/mask_util.py:19]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914&bid=19729641)

File: tools/ARIAtools/phaseMinimization.py (hunk 5 above)
Comment (dbekaert) [inline tools/ARIAtools/phaseMinimization.py:21]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/productPlot.py (hunk 1 above)
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/phaseMinimization.py (hunk 5 above)
Comment (dbekaert) [inline tools/ARIAtools/phaseMinimization.py:21]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/productPlot.py (hunk 1 above)
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/mask_util.py (hunk 4 above)
Comment (dbekaert) [inline tools/ARIAtools/mask_util.py:19]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/unwrapStitching.py [hunk 6]
```diff
@@ -27,29 +28,30 @@
 import glob
 import collections
 
+from ARIAtools.logger import logger
```
Comment (dbekaert) [inline tools/ARIAtools/unwrapStitching.py:31]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

Comment (dbekaert) [inline tools/ARIAtools/unwrapStitching.py:31]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/kml2box.py (hunk 3 above)
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/unwrapStitching.py [hunk 7]
```diff
@@ -62,6 +64,7 @@ def __init__(self):
         self.outFileConnComp = './connCompMerged'
         self.outputFormat='ENVI'
         self.verbose=False
+        log.setLevel('DEBUG') if self.verbose else ''
```
Comment (dbekaert) [inline tools/ARIAtools/unwrapStitching.py:67]: Codacy found an issue: [Expression "log.setLevel('DEBUG') if self.verbose else ''" is assigned to nothing](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/mask_util.py (hunk 4 above)
Comment (dbekaert) [inline tools/ARIAtools/mask_util.py:19]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/kml2box.py (hunk 3 above)
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6018915&bid=19729642)

File: tools/ARIAtools/computeMisclosure.py [hunk 8]
```diff
@@ -556,15 +559,15 @@ def __autoReferencePoint__(self):
 
             # Break loop after 10000 iterations
             if n == 10000:
-                print('WARNING: No reference point with coherence >= 0.7 found')
-                exit()
+                msg = f'No reference point with coherence >= {coh_min} found'
+                log.error(msg)
+                raise Exception(msg)
```
Comment (bbuzzanga) [inline tools/ARIAtools/computeMisclosure.py:564]: @Askaholic would you mind suggesting more pythonic way to do this?

File: tools/ARIAtools/unwrapStitching.py [hunk 9]
```diff
@@ -363,7 +366,7 @@ def __calculateCyclesOverlap__(self):
 
                 # determining the intersection between the two frames
                 if not bbox_frame1.intersects(bbox_frame2):
-                    print("Error, products do not overlap or were not provided in a contigious sorted list.")
+                    log.error("Products do not overlap or were not provided in a contigious sorted list.")
```
Comment (bbuzzanga) [inline tools/ARIAtools/unwrapStitching.py:368]: @Askaholic would you mind suggesting more pythonic way to do this, here as well?

File: tools/ARIAtools/computeMisclosure.py (hunk 8 above)
Comment (Askaholic) [reply tools/ARIAtools/computeMisclosure.py:564]: Hmm, I guess you could raise the exception here and then catch it higher up the call stack. Something like:

```python3
def the_function():
    if foo:
        raise Exception("Error message")

def main():
    try:
        the_function()
    except Exception:
        log.exception("An unexpected error occurred")
        # Note that log.exception also prints the exception trace including the exception message
```

Comment (bbuzzanga) [reply tools/ARIAtools/computeMisclosure.py:564]: @Askaholic won't this script continue to run if there are addition steps in main?

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity decreasing per file
==============================
+ tools/ARIAtools/computeMisclosure.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6018915&bid=19729642)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 15
           

Complexity decreasing per file
==============================
+ tools/ARIAtools/computeMisclosure.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914&bid=19729641)

File: tools/ARIAtools/mask_util.py (hunk 4 above)
Comment (dbekaert) [inline tools/ARIAtools/mask_util.py:19]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

Comment (dbekaert) [inline tools/ARIAtools/mask_util.py:19]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/phaseMinimization.py (hunk 5 above)
Comment (dbekaert) [inline tools/ARIAtools/phaseMinimization.py:21]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/computeMisclosure.py [hunk 10]
```diff
@@ -159,7 +164,7 @@ def __init__(self,imgfile,workdir='./',
         self.excludePairs = excludePairs
 
         # Other
-        self.verbose = verbose
+        log.setLevel('DEBUG') if self.verbose else ''
```
Comment (dbekaert) [inline tools/ARIAtools/computeMisclosure.py:167]: Codacy found an issue: [Expression "log.setLevel('DEBUG') if self.verbose else ''" is assigned to nothing](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/unwrapStitching.py (hunk 6 above)
Comment (dbekaert) [inline tools/ARIAtools/unwrapStitching.py:31]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/productPlot.py (hunk 1 above)
Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

Comment (dbekaert) [inline tools/ARIAtools/productPlot.py:15]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/unwrapStitching.py (hunk 7 above)
Comment (dbekaert) [inline tools/ARIAtools/unwrapStitching.py:67]: Codacy found an issue: [Expression "log.setLevel('DEBUG') if self.verbose else ''" is assigned to nothing](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/kml2box.py (hunk 3 above)
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/computeMisclosure.py [hunk 11]
```diff
@@ -805,9 +807,7 @@ def __misclosureQuery__(self,queryXY=None,queryLoLa=None):
             qx,qy = queryXY
             qLon,qLat = self.XY2LoLa(queryXY[0],queryXY[1])
 
-        if self.verbose == True:
-            print('Query point: X {} / Y {}; Lon {:.4f} / Lat {:.4f}'. \
-                format(qx,qy,qLon,qLat))
+        log.debug('Query point: X %s / Y %s; Lon %.4f / Lat %.4f', qx, qy, bqLon, qLat)
```
Comment (dbekaert) [inline tools/ARIAtools/computeMisclosure.py:810]: Codacy found an issue: [undefined name 'bqLon' (F821)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/unwrapStitching.py (hunk 6 above)
Comment (dbekaert) [inline tools/ARIAtools/unwrapStitching.py:31]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/computeMisclosure.py [hunk 12]
```diff
@@ -16,10 +16,15 @@
 import matplotlib.pyplot as plt
 import pandas as pd
 from pandas.plotting import register_matplotlib_converters
-register_matplotlib_converters()
 from osgeo import gdal
 from scipy.stats import mode
 
+import logging
+from ARIAtools.logger import logger
```
Comment (dbekaert) [inline tools/ARIAtools/computeMisclosure.py:23]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/phaseMinimization.py (hunk 5 above)
Comment (dbekaert) [inline tools/ARIAtools/phaseMinimization.py:21]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/kml2box.py (hunk 3 above)
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/computeMisclosure.py (hunk 12 above)
Comment (dbekaert) [inline tools/ARIAtools/computeMisclosure.py:23]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/computeMisclosure.py (hunk 8 above)
Comment (Askaholic) [reply tools/ARIAtools/computeMisclosure.py:564]: I would assume you basically just wrap the whole body of main in the try block. Actually what would be even better is if you wrapped the call to main in a try block. Then any python callers can catch the exceptions and handle them in their own way:

```python

if __name__ == '__main__':
    try:
        main()
    except Exception:
        log.exception('ditto')
```

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ tools/ARIAtools/computeMisclosure.py  -1
+ tools/ARIAtools/ARIAProduct.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6018915&bid=19729642)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity decreasing per file
==============================
+ tools/ARIAtools/computeMisclosure.py  -1
+ tools/ARIAtools/ARIAProduct.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914&bid=19729641)

File: tools/ARIAtools/kml2box.py (hunk 3 above)
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: [Unused logger imported from ARIAtools.logger](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/computeMisclosure.py [hunk 13]
```diff
@@ -805,9 +808,7 @@ def __misclosureQuery__(self,queryXY=None,queryLoLa=None):
             qx,qy = queryXY
             qLon,qLat = self.XY2LoLa(queryXY[0],queryXY[1])
 
-        if self.verbose == True:
-            print('Query point: X {} / Y {}; Lon {:.4f} / Lat {:.4f}'. \
-                format(qx,qy,qLon,qLat))
+        log.debug('Query point: X %s / Y %s; Lon %.4f / Lat %.4f', qx, qy, bqLon, qLat)
```
Comment (dbekaert) [inline tools/ARIAtools/computeMisclosure.py:811]: Codacy found an issue: [undefined name 'bqLon' (F821)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/kml2box.py (hunk 3 above)
Comment (dbekaert) [inline tools/ARIAtools/kml2box.py:13]: Codacy found an issue: ['ARIAtools.logger.logger' imported but unused (F401)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914)

File: tools/ARIAtools/computeMisclosure.py (hunk 8 above)
Comment (bbuzzanga) [reply tools/ARIAtools/computeMisclosure.py:564]: @Askaholic yep, agreed - from my reading this seems to be the best practice

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ tools/ARIAtools/ARIAProduct.py  -2
+ tools/ARIAtools/computeMisclosure.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6018915&bid=19729642)

**Comment by dbekaert** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ tools/ARIAtools/ARIAProduct.py  -2
+ tools/ARIAtools/computeMisclosure.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6018914&bid=19729641)

**Review by dbekaert — APPROVED** [review]
@bbuzz31 this looks good. I do not see a change wrt to the earlier PR we made, so will try to merge it and see if it overwrites anything. In case it does not then we should be good to go otherwise i can evert the merge.

**Pull request merged**